prebuild.py

```diff
@@ -0,0 +1,11 @@
+Import("env")
+
+# Set the built firmware binary to the correct name.
+binary_name = env.GetProjectOption("binary_name")
+env.Replace(PROGNAME=binary_name)
+
+# Allow the name of the binary to be reported by the webapp
+env.Append(CPPDEFINES=[
+  ("PIO_BINARY", binary_name)
```

Comment:
I don't think PIO_BINARY would be set for someone who compiles the sketch in the Arduino IDE, right? I mean, the clue is in the name 😄 

I'm not a maintainer on this project, just an interested bystander - but instead of displaying a filename (which can change in future updates if someone decides to change the naming structure), WDYT about just showing the board info? It looks like the preprocessor gets a couple of helpful defines at compile time - here's what's generated when compiling in the IDE for my D1 Mini :
```
"-DARDUINO_BOARD=\"ESP8266_WEMOS_D1MINI\""
"-DARDUINO_BOARD_ID=\"d1_mini_clone\""
```

I got this by enabling verbose compile output in the Arduino IDE. It looks like these are set in platformio builds as well? Grepping for `ARDUINO_BOARD_ID` in my Arduino libraries folder looks promising:

```
❯ rg ARDUINO_BOARD_ID
packages/esp8266/hardware/esp8266/3.1.1/libraries/ESP8266mDNS/src/LEAmDNS.cpp
35:#ifndef ARDUINO_BOARD_ID
36:#define ARDUINO_BOARD_ID "generic"
1275:                || (!addServiceTxt(hService, "board", ARDUINO_BOARD_ID))
 
packages/esp8266/hardware/esp8266/3.1.1/platform.txt
131:recipe.c.o.pattern="{compiler.path}{compiler.c.cmd}" {compiler.cpreprocessor.flags} {compiler.c.flags} -D{build.sdk}=1 -DF_CPU={build.f_cpu} {build.lwip_flags} {build.debug_port} {build.debug_level} -DA>
134:recipe.cpp.o.pattern="{compiler.path}{compiler.cpp.cmd}" {compiler.cpreprocessor.flags} {compiler.cpp.flags} -D{build.sdk}=1 -DF_CPU={build.f_cpu} {build.lwip_flags} {build.debug_port} {build.debug_leve>
137:recipe.S.o.pattern="{compiler.path}{compiler.c.cmd}" {compiler.cpreprocessor.flags} {compiler.S.flags} -D{build.sdk}=1 -DF_CPU={build.f_cpu} {build.lwip_flags} {build.debug_port} {build.debug_level} -DA>
 
packages/esp8266/hardware/esp8266/3.1.1/tools/platformio-build.py
130:        ("ARDUINO_BOARD_ID", '\\"%s\\"' % env.BoardConfig().id),
```

I feel like displaying the board info would meet the need of "remind me what piece of hardware this is" without requiring any build hacking or prebuild scripts. I do think it's a great idea to display board info in the UI!

Reply:
Oh interesting, I had no idea it was even possible to build this in Arduino ide (I've never used that before).

I did see those defines when I had verbose build enabled myself, but personally I didn't think they gave enough info, doesn't show which build profile was actually used etc.

Best of both however would be to use the PIO define if it exists, else the C code could fallback to the Arduino define.

Reply:
sure, doing both makes sense to me! seems like they could just be on separate lines. there’s plenty of room on that page. 

filename: `PIO_BINARY` | (not set)
target: `ARDUINO_BOARD` | (not set)

Reply:
> Oh interesting, I had no idea it was even possible to build this in Arduino ide (I've never used that before).

it does seem like I need to get hip to using PlatformIO…I’m a total dabbler in the embedded world 🤣